clubhouse/api.py

```diff
@@ -0,0 +1,232 @@
+
+        Args:
+            method (str): Must be "get", "post", "put" or "delete"
+            id (int): The requested item's ID
+        Returns:
+            A JSON object containing information about the requested item
+        '''
+        result = self._request("get", None, *segments, **kwargs)
+        return result.json()
+
+    def _list_items(self, *segments, **kwargs):
+        '''An internal helper method for calling any "List"-related endpoint.
+
+        Args:
+            method (str): Must be "get"
+        Returns:
+            A list of dictionaries, where each dictionary is one item
+        '''
+        result = self._request("get", None, *segments, **kwargs)
+        items = result.json()
+        while result.next:
+            result = self._request("get", None, result.next, **kwargs)
+            items.extend(result.json())
+        return items
+
+    def _update_item(self, data, *segments, **kwargs):
+        '''An internal helper method for calling any "Update"-related endpoint.
+
+        Args:
+            data (dict): Can contain any of the body parameters listed in the
+                API reference for the endpoint.
+        Returns:
+            A JSON object containing information about the updated item
+        '''
+        result = self._request("put", data, *segments, **kwargs)
+        return result.json()
+
+    ################
+    #  Milestones  #
+    ################
+
+    def create_milestone(self, data, **kwargs):
+        '''Create a Milestone.
+        https://clubhouse.io/api/rest/v3/#Create-Milestone
+
+        Example:
+            from clubhouse import ClubhouseClient
+            conn = ClubhouseClient(API_KEY)
+            conn.create_milestone({'name': 'TEST'})
+
+        Args:
+            data (dict): Can contain any of the body parameters listed in the
+                API reference linked above as keys.
+        Returns:
+            A JSON object containing information about the new Milestone.
+        '''
+        segments = ["milestones"]
+        return self._create_item(data, *segments, **kwargs)
+
+    def delete_milestone(self, id, **kwargs):
+        '''Delete a specific Milestone.
+        https://clubhouse.io/api/rest/v3/#Delete-Milestone
+
+        Example:
+            from clubhouse import ClubhouseClient
+            conn = ClubhouseClient(API_KEY)
+            conn.delete_milestone(123)
+
+        Args:
+            id (int): The Milestone ID
+
+        Returns:
+            An empty dictionary
+        '''
+        segments = ["milestones", id]
+        return self._delete_item(*segments, **kwargs)
+
+    def get_milestone(self, id, **kwargs):
+        '''Retrieve a specific Milestone.
+        https://clubhouse.io/api/rest/v3/#Get-Milestone
+
+        Example:
+            from clubhouse import ClubhouseClient
+            conn = ClubhouseClient(API_KEY)
+            conn.get_milestone(123)
+
+        Args:
+            id (int): The Milestone ID
```

Comment:
I would personally add type hints (not that I did in the previous version), although that would impact compatibility with older versions of python.

Reply:
Ah, interesting. I will look into this angle, thanks!

Reply:
Still looking into this, but resolving for now. May add this in.